transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -365,7 +385,11 @@ public ChannelFuture writeOneInbound(Object msg) {
      */
     public ChannelFuture writeOneInbound(Object msg, ChannelPromise promise) {
         if (checkOpen(true)) {
-            pipeline().fireChannelRead(msg);
+            if (!config().isAutoRead()) {
+                tempInboundMessages().add(msg);
+            } else {
+                pipeline().fireChannelRead(msg);
```

Comment:
is this correct ? Dont we need to drain the inbound messages first to ensure we preserve the correct order ?

Reply:
you are right. I made changes. By the way, because `writeOneInbound` may have concurrent calls, I modified the type of `tempInboundMessages` to make it thread-safe.

Reply:
I still think we may need to empty the queue before we fire this message through the pipeline as otherwise it may produce out of order message sequences.

Reply:
This is causing me some confusion.
It seems that when `config ().isAutoRead() = true`,` tempInboundMessages` must have been cleared.
Can you explain under what circumstances will the disorder occur?

Reply:
you are right... maybe want to add an assert ? Sorry for the confusion.

Reply:
It doesn't matter. :)

Agree, I will fix other review requests later.